### PR TITLE
Connectivity: compute_matrices script

### PR DESCRIPTION
**Pull request by EmmaRenauld**
Refactored the scil_connectivity_compute_matrices.py script:

- Took out the main computation part and put it in `connectivity.connectivity` file
- (also renamed `connectivity.connectivity_tools` to `connectivity.matrix_too0`ls)
- Clarified the docstring
    - Note that I already mention the `scil_connectivity_compute_simple_matrix.py`, which does not exist, but in the end I think that this PR is big enough. Will add it soon in another PR.
    -  Please help me describe the --similarity option better. 
- Removed the `--maps` option, as discussed with François and Arnaud offline.
- Removed one buggy logging.info: the `-len(label_list)` could end up in a negative number of total_elem if some diagonal values were not found.

image

- Now all headers compatibility are done before loading (except for the --similarity folder)

In short, the code is simplified a lot, but the result should be exactly the same, if you want to test with some data!

### CODE REVIEW

**Review by frheault — COMMENTED** [review]
I used the new script on older data to compare before vs after.

My command was `scil_connectivity_compute_matrices.py Transform_Data/sub-001_ses-1__decompose_warped_mni.h5 Transform_Data/sub-001_ses-1__labels_warped_mni_int16.nii.gz --volume vol.npy --streamline_count sc.npy --length len.npy --metrics Transform_Metrics/sub-001_ses-1__fa_mni.nii.gz fa.npy --include_dps dps -f` (works with the master)

This is my error:
```

Traceback (most recent call last):
  File "/home/rhef1902/VirtualEnvs/basic_scilpy_3_10/bin/scil_connectivity_compute_matrices.py", line 33, in <module>
    sys.exit(load_entry_point('scilpy', 'console_scripts', 'scil_connectivity_compute_matrices.py')())
  File "/home/rhef1902/Libraries/scilpy/scripts/scil_connectivity_compute_matrices.py", line 260, in main
    outputs.append(compute_connectivity_matrices_from_hdf5(
  File "/home/rhef1902/Libraries/scilpy/scilpy/connectivity/connectivity.py", line 188, in compute_connectivity_matrices_from_hdf5
    for dps_key in hdf5_file[key].keys():
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/home/rhef1902/VirtualEnvs/basic_scilpy_3_10/lib/python3.10/site-packages/h5py/_hl/group.py", line 357, in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5o.pyx", line 189, in h5py.h5o.open
KeyError: 'Unable to synchronously open object (invalid identifier type to function)'
```
I am one Python 3.10.12 (and I got the same error in Python 3.11.6)

**Comment by EmmaRenauld** [comment]
> I used the new script on older data to compare before vs after.

Could you please send me your data? Thanks!

File: scripts/scil_connectivity_compute_matrices.py [hunk 1]
```diff
@@ -3,273 +3,133 @@
...
+                             validate_nbr_processes, assert_inputs_dirs_exist,
+                             assert_headers_compatible,
+                             assert_output_dirs_exist_and_empty)
 
 
 def _build_arg_parser():
     p = argparse.ArgumentParser(
         description=__doc__,
-        formatter_class=argparse.RawTextHelpFormatter,)
+        formatter_class=argparse.RawTextHelpFormatter, )
```
Comment (arnaudbore) [inline scripts/scil_connectivity_compute_matrices.py:89]: you can remove the comma here

File: scilpy/connectivity/connectivity.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+def load_node_nifti(directory, in_label, out_label, ref_img):
+    in_filename = os.path.join(directory,
+                               '{}_{}.nii.gz'.format(in_label, out_label))
+
+    if os.path.isfile(in_filename):
+        if not is_header_compatible(in_filename, ref_img):
+            raise IOError('{} do not have a compatible header'.format(
+                in_filename))
+        return nib.load(in_filename).get_fdata(dtype=np.float64)
```
Comment (arnaudbore) [inline scilpy/connectivity/connectivity.py:34]: float32 enough ?

File: scilpy/connectivity/connectivity.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        If true, return 'volume' in the returned dictionary with the volume of
+        the bundle.
+    compute_streamline_count: bool
+        If true, return 'streamline_count' in the returned dictionary, with
+        the number of streamlines in the bundle.
+    compute_length: bool
+        If true, return 'length' in the returned dictionary, with the mean
+        length of streamlines in the bundle.
+    similarity_directory: str
+        If not None, ??
```
Comment (arnaudbore) [inline scilpy/connectivity/connectivity.py:78]: ?? :D

File: scilpy/connectivity/connectivity.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            len(metrics_data) > 0):
+        density = compute_tract_counts_map(streamlines, dimensions)
+
+    measures_to_return = {}
+
+    if compute_length:
+        # scil_tractogram_segment_connections_from_labels.py requires
+        # isotropic voxels
+        mean_length = np.average(length(list(streamlines))) * voxel_sizes[0]
+        measures_to_return['length'] = mean_length
```
Comment (arnaudbore) [inline scilpy/connectivity/connectivity.py:134]: should we explicit length_mm ?

File: scilpy/connectivity/connectivity.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+    if compute_volume:
+        measures_to_return['volume'] = np.count_nonzero(density) * \
```
Comment (arnaudbore) [inline scilpy/connectivity/connectivity.py:137]: _mm3 ?

**Review by arnaudbore — APPROVED** [review]
LGTM !

File: scilpy/connectivity/connectivity.py (hunk 2 above)
Comment (EmmaRenauld) [reply scilpy/connectivity/connectivity.py:34]: @frheault ? I changed it to 32 as per this suggestion. Is it ok?

File: scilpy/connectivity/connectivity.py (hunk 3 above)
Comment (EmmaRenauld) [reply scilpy/connectivity/connectivity.py:78]: @frheault  I wasn't sure how to explain this value. Can you help please?

**Comment by EmmaRenauld** [comment]
> This is my error:

Sorry @frheault . Fixed!

File: scilpy/connectivity/connectivity.py (hunk 3 above)
Comment (frheault) [reply scilpy/connectivity/connectivity.py:78]: It is a directory with nifti that represents density maps for each connection using the {in_label}_{out_label}.nii.gz conventions. Typically computed from a template (must be in the same space)
